modules/islandora_binary_object_zip_importer/includes/importer.inc

```diff
@@ -0,0 +1,257 @@
+    $ns_options = array(
+      NULL => t('Default namespace'),
+    );
+    // By this point we should know that the collection has a policy with a
+    // binary object content model.
+    $policy = new CollectionPolicy($parent['COLLECTION_POLICY']->content);
+    $models = $policy->getContentModels();
+    $namespace = $models['islandora:binaryObjectCModel']['namespace'];
+    if (!empty($namespace)) {
+      module_load_include('inc', 'islandora', 'includes/utilities');
+      $namespace = islandora_get_namespace($namespace);
+      $ns_options[$namespace] = $namespace;
+    }
+
+    return array(
+      'fs' => array(
+        '#type' => 'fieldset',
+        '#title' => t('Binary Object Zip Batch Importer'),
+        '#description' => t('Select the file containing the assets and metadata to import. Assets and metadata will be matched together based on the portion of the filename without the extension, e.g., my_file.xml and my_file.pdf would be combined into a single object. Only one file should be paired with each .xml, if this is the case. This importer currently does not support importing stub objects with .xml records.'),
+        'file' => array(
+          '#type' => 'managed_file',
+          '#title' => t('Zip file containing files to import'),
+          '#upload_validators' => array(
+            'file_validate_extensions' => array('zip'),
+          ),
+        ),
+        'namespace' => array(
+          '#type' => 'select',
+          '#title' => t('Object Namespace'),
+          '#description' => t('The namespace in which the imported objects will be created.'),
+          '#options' => $ns_options,
+        ),
+        'submit' => array(
+          '#type' => 'submit',
+          '#value' => t('Import'),
+        ),
+      ),
+    );
+  }
+
+  /**
+   * Determine if we're ready to process the batch.
+   *
+   * @see IslandoraBatchImporter::readyForBatch()
+   */
+  public static function readyForBatch(array &$form_state) {
+    if (empty($form_state['values']['file'])) {
+      form_set_error('file', t('Need a Zip file!'));
+      return FALSE;
+    }
+    return TRUE;
+  }
+
+  /**
+   * Get info for the source parameter for the itemClass factory.
+   *
+   * @see IslandoraBatchImporterInterface::getBatchInfo()
+   */
+  public static function getBatchInfo(array &$form_state) {
+    // Can be either a path or a file object ID.
+    if (!file_exists($form_state['values']['file'])) {
+      $file = file_load($form_state['values']['file']);
+    }
+    else {
+      $file = $form_state['values']['file'];
+    }
+    return array(
+      'file' => $file,
+      'pid_namespace' => $form_state['values']['namespace'],
+      'object_info' => static::getIngestInfo($file),
+      'processed_objects' => array(),
+    );
+  }
+
+  /**
+   * Helper for getting info from the zip file.
+   *
+   * @param stdClass|string $file
+   *   A file object or path to a zip file.
+   */
+  protected static function getIngestInfo($file) {
```

Comment:
So it can support being given the path to a file from the Drush version as well as a file object from the UI version